shell/platform/windows/platform_handler.cc

```diff
@@ -372,30 +382,31 @@ void PlatformHandler::SystemExitApplication(
   }
 }
 
-void PlatformHandler::RequestAppExit(const std::string& exit_type,
-                                     int64_t exit_code) {
+void PlatformHandler::RequestAppExit(ExitType exit_type, UINT exit_code) {
   auto callback = std::make_unique<MethodResultFunctions<rapidjson::Document>>(
       [this, exit_code](const rapidjson::Document* response) {
         RequestAppExitSuccess(response, exit_code);
       },
       nullptr, nullptr);
   auto args = std::make_unique<rapidjson::Document>();
   args->SetObject();
-  args->GetObjectW().AddMember(kExitTypeKey, exit_type, args->GetAllocator());
+  args->GetObjectW().AddMember(
+      kExitTypeKey, std::string(exit_type_names[static_cast<int>(exit_type)]),
+      args->GetAllocator());
   channel_->InvokeMethod(kRequestAppExitMethod, std::move(args),
                          std::move(callback));
 }
 
 void PlatformHandler::RequestAppExitSuccess(const rapidjson::Document* result,
-                                            int64_t exit_code) {
+                                            UINT exit_code) {
   const std::string& exit_type = result[0][kExitResponseKey].GetString();
```

Comment:
We cannot assume that documents are well-formed as that can cause crashes if that assumption is incorrect. We should ignore malformed messages but log an error.